stac_fastapi/api/stac_fastapi/api/config.py

```diff
@@ -22,3 +28,48 @@ class AddOns(enum.Enum):
     """Enumeration of available third party add ons."""
 
     bulk_transaction = "bulk-transaction"
+
+
+def env_to_sequence(
+    env_var: str, default: Sequence[str], sequence_separator: str = "|"
+) -> Sequence[str]:
+    """Retrieve a sequence of values from an env var string, or default if missing."""
+    if env_var in environ:
+        if re.search(re.escape(sequence_separator), environ[env_var]):
+            return tuple(
+                [part for part in environ[env_var].split(sequence_separator) if part]
+            )
+        else:
+            return (environ[env_var],)
+    else:
+        return default
+
+
+def env_to_str(env_var: str, default: str) -> str:
+    """Retrieve a string from an env var, or default if missing."""
+    if env_var in environ:
+        return environ[env_var]
+    else:
+        return default
+
+
+def env_to_bool(env_var: str, default: bool) -> bool:
+    """Retrieve a bool from an env var, or default if missing."""
+    if env_var in environ:
+        if re.match("^(true|1)$", environ[env_var], re.IGNORECASE):
+            return True
+        if re.match("^(false|0)$", environ[env_var], re.IGNORECASE):
+            return False
+        logger.warning(f"{env_var} set but not a valid bool")
+    return default
+
+
+def env_to_int(env_var: str, default: int) -> int:
+    """Retrieve an int from an env var, or default if missing."""
+    if env_var in environ:
+        value = environ[env_var].strip()
+        if value.isdigit():
+            return int(value)
+        else:
+            logger.warning(f"{env_var} set but not a valid int")
+    return default
```

Comment:
could we just use `pydantic.BaseSettings` https://pydantic-docs.helpmanual.io/usage/settings/

It should take care of everything needed

Reply:
Yes, sorry, I just had that realisation a few minutes earlier